### ISSUE TITLE
Ground Truth is used

### ISSUE THREAD

**Issue by bio-mlhui** [hynnsk/HP#12]
I saw that the ground truth label is using when training:

```python
model_output = net_model(img, train=True)
...
detached_code = torch.clone(model_output[1].detach())
linear_output = linear_model(detached_code)
...
linear_loss = self.linear_loss(linear_output, label, self.n_classes)
loss = linear_loss + cluster_loss
...
...
scaler.scale(loss).backward()
```
In my opinion, alghough `detached_code ` does not requires gradient. The `linear_output ` requires gradient and will generate a new torch graph. The `linear_loss` will finally backward gradient to the `linear_model`. 
Also the evaluate code is:
```python
output = net_model(img)
...
linear_preds = torch.log_softmax(linear_model(output[1]), dim=1)
...
linear_metrics.update(linear_preds, label)
...
eval_metrics = get_metrics(cluster_metrics, linear_metrics)
```
As shown, the `linear_model` is used when computing the 'Linear_' metrics. 

Does this mean the ground truth pixel-wise mask label is used when training and HP is not unsupervised? 

I am very confused. It would be very kind for you to teach me where I am wrong. Thanks a lot!

**Comment by hynnsk** [comment]
Hello,

Thank you for your interest in our work.

The linear model and cluster model exist solely for performance evaluation purposes.

In specific, segmentation features are detached to train a linear classifier and cluster centers.
Linear classifier is trained with ground-truth labels and cluster centers are trained with running k-means algorithm.

The performance evaluation using linear classifier is commonly used in self-supervised learning tasks. Example papers are as follows:

- Deep Clustering for Unsupervised Learning of Visual Features
- A Simple Framework for Contrastive Learning of Visual Representations
- Unsupervised Learning of Visual Features by Contrasting Cluster Assignments
- Emerging Properties in Self-Supervised Vision Transformers

**Comment by Xu-Huihui** [comment]
Quite helpful. I have figured out the 'linear probe' thing. Thanks so much.
